src/shared/components/wallet/Wallet.js

```diff
@@ -0,0 +1,81 @@
+import React, { Component } from 'react';
+
+import createWallet from 'idm-wallet';
+import { CSSTransition } from 'react-transition-group';
+
+import WalletContent from './wallet-content/WalletContent';
+import SetupLocker from '../setup-locker/SetupLocker';
+import LockScreen from '../lock-screen';
+
+import styles from './Wallet.css';
+
+const lockscreenClassNames = {
+    enter: styles.lockscreenEnter,
+    enterActive: styles.lockscreenEnterActive,
+    exit: styles.lockscreenExit,
+    exitActive: styles.lockscreenExitActive,
+};
+
+class Wallet extends Component {
+    getLock = undefined;
+
+    state = {
+        wallet: undefined,
+    };
+
+    componentDidMount() {
+        createWallet().then((wallet) => {
+            wallet.locker.onLockedChange(this.handleLockedChanged);
+            this.getLock = wallet.locker.getLock.bind(wallet.locker);
+
+            this.setState({ wallet });
+        });
+    }
+
+    render() {
+        const { wallet } = this.state;
+
+        if (wallet) {
+            return this.renderWallet();
+        }
+
+        return <div>No Wallet</div>;
+    }
+
+    renderWallet = () => {
+        const { wallet } = this.state;
+        const { locker } = wallet;
+
+        if (locker.isPristine()) {
+            return <SetupLocker locker={ locker } onComplete={ this.handleSetupLockerComplete } />;
+        }
+
+        const isLocked = locker.isLocked();
+
+        return (
+            <div className={ styles.parent }>
+                <CSSTransition classNames={ lockscreenClassNames } in={ isLocked }
+                    mountOnEnter unmountOnExit timeout={ 1800 }>
+                    <LockScreen getLock={ this.getLock } unmounting={ !isLocked } />
+                </CSSTransition>
+                <WalletContent wallet={ wallet } />
```

Comment:
Related to my previous comment, maybe everything below this Wallet Content shouldn't be commited and only kept for demo purposes?